### PR TITLE
Volume endpoint changes/tweaks

### PR DESCRIPTION
**Pull request by TristanWright**

### CODE REVIEW

**Comment by TristanWright** [comment]
- [ ] fix delete_volume test
- [ ] add sse test

**Comment by TristanWright** [comment]
when running the delete test the volume status is not being updated to available, huh

File: girder/cumulus/server/models/volume.py [hunk 1]
```diff
@@ -109,6 +111,16 @@ def append_to_log(self, user, id, record):
         self.update({'_id': ObjectId(id)}, {'$push': {'log': record}})
         send_log_notification('volume', volume, record)
 
+    def update_volume(self, user, volume):
```
Comment (cjh1) [inline girder/cumulus/server/models/volume.py]: I'm not sure this should be call update_volume, its not really updating the existing volume, its more overriding it. I would be tempted to call it save_volume. The updating code is really in the patch method.

File: girder/cumulus/server/volume.py [hunk 2]
```diff
@@ -64,14 +64,13 @@ def __init__(self):
         self._model = self.model('volume', 'cumulus')
 
     def _create_ebs(self, body, zone):
-        user = self.getCurrentUser()
```
Comment (cjh1) [inline girder/cumulus/server/volume.py]: What did this do?

File: girder/cumulus/server/models/volume.py (hunk 1 above)
Comment (TristanWright) [inline girder/cumulus/server/models/volume.py]: Should I be calling save in some places then? `self.update(volume)` was not clearing the volume's `clusterId` after a detach_complete (and so it couldn't be deleted)

File: girder/cumulus/server/volume.py (hunk 2 above)
Comment (TristanWright) [inline girder/cumulus/server/volume.py]: We're importing getCurrentUser() above, isn't the self redundant?

File: girder/cumulus/server/models/volume.py (hunk 1 above)
Comment (TristanWright) [inline girder/cumulus/server/models/volume.py]: fixed this by amending last commit, server/volume.py uses send_status_notification for detach_complete

File: girder/cumulus/server/volume.py [hunk 3]
```diff
@@ -204,8 +204,8 @@ def find(self, params):
                level=AccessType.ADMIN)
     @loadmodel(model='volume', plugin='cumulus', level=AccessType.ADMIN)
     def attach_complete(self, volume, cluster, params):
+        user = self.getCurrentUser()
```
Comment (cjh1) [inline girder/cumulus/server/volume.py]: Shouldn't this be getCurrentUser()?

File: girder/cumulus/server/volume.py [hunk 4]
```diff
@@ -342,18 +341,18 @@ def detach(self, volume, params):
         if master['state'] != InstanceState.RUNNING:
             raise RestException('Master instance is not running!',
                                 400)
-
+        user = self.getCurrentUser()
```
Comment (cjh1) [inline girder/cumulus/server/volume.py]: getCurrentUser()?

File: girder/cumulus/server/volume.py [hunk 5]
```diff
@@ -367,8 +366,9 @@ def detach(self, volume, params):
     def detach_complete(self, volume, params):
 
         # First remove from cluster
+        user = self.getCurrentUser()
```
Comment (cjh1) [inline girder/cumulus/server/volume.py]: getCurrentUser()?

**Pull request merged**